python_version: 3.5.0
failfast: yes
always run:
  - import sys

  - |
    class ExampleException(Exception):
        """
        This is a demonstration exception's docstring.

        It spreads across multiple lines.
        """
        pass

  - |
    def should_run(which):
        with open("should{0}.txt".format(which), "w") as handle:
            handle.write("ran!")

  - |
    def should_not_run():
        raise RuntimeError("This shouldn't have happened")

  - |
    def raise_example_exception():
        raise ExampleException

  - |
    def output(contents):
        with open("output.txt", 'a') as handle:
            handle.write("{0}\n".format(contents))
